File: sakuralearn-frontend/src/services/noteService.js
import API from './api'

export const noteService = {
  createNote: (data) => API.post('/notes', data),
  updateNote: (noteId, content) => API.put(`/notes/${noteId}`, content, { headers: { 'Content-Type': 'text/plain' } }),
  deleteNote: (noteId) => API.delete(`/notes/${noteId}`),
  getNotesByBlock: (blockId) => API.get(`/notes/block/${blockId}`),
  getNotesByLesson: (lessonId) => API.get(`/notes/lesson/${lessonId}`),
}

File: sakuralearn-backend/src/main/java/com/sakuralearn/sakuralearn_backend/service/NoteService.java
package com.sakuralearn.sakuralearn_backend.service;

import com.sakuralearn.sakuralearn_backend.dto.request.PersonalNoteRequest;
import com.sakuralearn.sakuralearn_backend.dto.response.PersonalNoteResponse;

import java.util.List;
import java.util.UUID;

public interface NoteService {
    PersonalNoteResponse createNote(UUID userId, PersonalNoteRequest request);
    PersonalNoteResponse updateNote(UUID userId, UUID noteId, String content);
    void deleteNote(UUID userId, UUID noteId);
    List<PersonalNoteResponse> getNotesByBlock(UUID userId, UUID blockId);
    List<PersonalNoteResponse> getNotesByLesson(UUID userId, UUID lessonId);
}
